sync3/handler/connstate.go

```diff
@@ -624,6 +624,7 @@ func (s *ConnState) getInitialRoomData(ctx context.Context, roomSub sync3.RoomSu
 			JoinedCount:       metadata.JoinCount,
 			InvitedCount:      &metadata.InviteCount,
 			PrevBatch:         userRoomData.RequestedLatestEvents.PrevBatch,
+			Timestamp:         metadata.LastMessageTimestamp,
```

Comment:
You don't know if this is what they want. `getInitialRoomData` is a central point called for multiple lists: you need to know the `bump_event_types` values to know what timestamp to pull here. Add that as a param to `getInitialRoomData`